corehq/form_processor/reprocess.py

```diff
@@ -48,7 +48,14 @@ def reprocess_unfinished_stub(stub, save=True):
         return
 
     assert form.is_normal
-    result = _reprocess_form(form, save)
+    try:
+        result = _reprocess_form(form, save)
+    except:
+        if save:
+            stub.queued = False
+            stub.save()
```

Comment:
@gcapalbo I've updated to use `enqueuing_timeout` but have still kept the limit on the query since the query is also filtered by the same parameters.